src/tests/uat/recommendation_uat.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020-2021. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""
+The following API calls are tested in this script.
+
+For the validation CBC API requests are used.
+
+To execute, a profile must be provided using the standard CBC Credentials.
+
+Recommendations:
+* Search Recommendations
+
+The following API calls were not covered:
+* Recommendation Workflow
+"""
+
+import sys
+import requests
+from cbc_sdk.helpers import build_cli_parser, get_cb_cloud_object
+from cbc_sdk.endpoint_standard import Recommendation
+
+
+def compare_impact(impact, raw_data):
+    """
+    Compare a RecommendationImpact object against the data that should represent it.
+
+    Args:
+        impact (RecommendationImpact): The RecommendationImpact object to compare.
+        raw_data (dict): The dictionary containing what should be the same data.
+    """
+    assert impact.event_count == raw_data['event_count']
```

Comment:
Please, add error messages to be displayed in the console upon failure instead of a thrown error (for all asserts).

Reply:
I had to rewrite most of the script, but I think I've given you what you want.

Reply:
The good thing now is that the test will go through all the checks before exiting but in case of failure it won't tell you which check failed which means failures will take more time to investigate. The initial way you proposed seems more readable. What do you think about using:
assert <expression>, <error_message>
It will end the test on the first failure but will give more user-friendly error and will be more readable.